src/app/business/profile/profile.component.ts

```diff
@@ -68,12 +72,51 @@ export class ProfileComponent implements OnInit {
     }
     getProfiles() {
         this.ProfileService.getProfiles().subscribe((res) => {
-            this.profiles = res.json();
-            this.profiles.forEach(item=>{
+            let str = res.json();
+            this.allProfileNameForCheck = [];
+            str.forEach(item=>{
                 if(item.storageType && item.storageType == "file"){
                     delete item.replicationProperties;
                 }
+                this.allProfileNameForCheck.push(item.name);
             })
+            this.getPools(str);
         });
     }
+    getPools(str) {
+        this.PoolService.getPools().subscribe((res) => {
+            this.pools = res.json();
+            str.forEach(item => {
+                item.totalFreeCapacity = this.getSumCapacity(this.pools, 'free',item);
+                item.totalCapacity = this.getSumCapacity(this.pools, 'total',item);
+            });
+            this.profiles = str;
+        },
+        err=>{
+            this.profiles = str;
+        });
+    }
+    getSumCapacity(pools, FreeOrTotal,data) {
+        let SumCapacity: number = 0;
+        let newPools = lodash.cloneDeep(pools);
+        newPools = newPools.filter(item=>{
+            return item.storageType == data.storageType;
```

Comment:
as commented above, please use a new API to query pools by storage type, use those pools here

Reply:
As above, this may be beyond the scope of this PR.

Reply:
LG...AB
Please add this and previous comment to backlog, thanks

Reply:
Added in issue #189

Reply:
LGTM